### PR TITLE
Updates JSON ABI, LDC, BSIZ, BLDD and ED19.

### PR DESCRIPTION
**Pull request by esdrubal**
## Description

Updates all the dependencies for the current release.

Implements the fuel ABI generation changes proposed in https://github.com/FuelLabs/fuel-specs/pull/599.

Removes the flag `--json-abi-with-callpaths` and the behavior is as if it were true. We removed the flag because it is unsafe to produce JSON ABIs without callpaths, so we shouldn't allow it.

Includes the LDC, BSIZ, BLDD and ED19 changes from:https://github.com/FuelLabs/sway/pull/6409.

Fixes https://github.com/FuelLabs/sway/issues/5954
Fixes https://github.com/FuelLabs/sway/issues/5151

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by sdankel — COMMENTED** [review]
Hey @esdrubal I'm getting caught up on this. It's probably too late now to change this, but I'm wondering, what's the value of hashing the string? It seems like it's more useful to leave the string un-hashed (will full paths for uniqueness) and human readable.

**Comment by kayagokalp** [comment]
I think we have an issue.

We need to have the proxy contract implementation's abi, storage slots, etc built and embedded in the code. This used to work until now. Problem is:

1. Due to breaking changes, we need to rebuild the proxy contract (audited one from sway-standard-implementations) but the namespace syntax is removed with the current version. 
2. I don't think it is related to the syntax but something changed about how storage key slots are assigned, the storage slot embedded into the sway-libs project is wrong (we used to get `0x7bb458adc1d118713319a5baa00a2d049dd64d2916477d2688d76970c898cd55` but now we are getting something else in the generated file, the lib value however stays the same) And built/released contract is using that key to interact with `TARGET` field of the storage.


## Summary of the problem

1. Before breaking changes, we had a bin file that was working nicely. Now due to the breaking changes (probably around encoding) we need to rebuilt contract to get the new bin, otherwise we are getting out of bounds panic from sdk while trying to assign configurables.  This is expected as change in offsets create a problem while inserting the configurable value into the bytecode.
2. Once we do that, we are facing with a new issue which is removal of `#[namepsace(SRC14)]`. This can be fixed by converting the storage declaration from
```sway
#[namepsace(SRC14)]
storage {
    /// The [ContractId] of the target contract.
    ///
    /// # Additional Information
    ///
    /// `target` is stored at sha256("storage_SRC14_0")
    target: Option<ContractId> = None,
    /// The [State] of the proxy owner.
    ///
    /// # Additional Information
    ///
    // `proxy_owner` is stored at sha256("storage_SRC14_1")
    proxy_owner: State = State::Uninitialized,
}
```
to
```sway
storage {
    SRC14 {
    /// The [ContractId] of the target contract.
    ///
    /// # Additional Information
    ///
    /// `target` is stored at sha256("storage_SRC14_0")
    target: Option<ContractId> = None,
    /// The [State] of the proxy owner.
    ///
    /// # Additional Information
    ///
    // `proxy_owner` is stored at sha256("storage_SRC14_1")
    proxy_owner: State = State::Uninitialized,
    }
}
```
3. Once we do that we come to the last issue, target's storage field is now different instead of the original value (`0x7bb458adc1d118713319a5baa00a2d049dd64d2916477d2688d76970c898cd55`) and since audited proxy contract takes an harcoded storage key from `sway-libs` this is not correct and execution fails.



## TL DR

The following two implementations before and after `#[namespace]` is removed, does not yield same storage slot keys. If this is something expected no problems otherwise might be important to look at.

```sway
#[namepsace(SRC14)]
storage {
    /// The [ContractId] of the target contract.
    ///
    /// # Additional Information
    ///
    /// `target` is stored at sha256("storage_SRC14_0")
    target: Option<ContractId> = None,
    /// The [State] of the proxy owner.
    ///
    /// # Additional Information
    ///
    // `proxy_owner` is stored at sha256("storage_SRC14_1")
    proxy_owner: State = State::Uninitialized,
}
```
```sway
storage {
    SRC14 {
    /// The [ContractId] of the target contract.
    ///
    /// # Additional Information
    ///
    /// `target` is stored at sha256("storage_SRC14_0")
    target: Option<ContractId> = None,
    /// The [State] of the proxy owner.
    ///
    /// # Additional Information
    ///
    // `proxy_owner` is stored at sha256("storage_SRC14_1")
    proxy_owner: State = State::Uninitialized,
    }
}
```

## Solution

I also found the solution, basically changing the storage declaration to the following fixes everything:

```sway
storage {
    /// The [ContractId] of the target contract.
    ///
    /// # Additional Information
    ///
    /// `target` is stored at sha256("storage_SRC14_0")
    target in 0x7bb458adc1d118713319a5baa00a2d049dd64d2916477d2688d76970c898cd55: Option<ContractId> = None,
    /// The [State] of the proxy owner.
    ///
    /// # Additional Information
    ///
    // `proxy_owner` is stored at sha256("storage_SRC14_1")
    proxy_owner: State = State::Uninitialized,
}
```
Though with this change I am touching the audited contract, is this ok?

https://github.com/FuelLabs/sway/pull/6416 is a dummy PR I opened to show that pinning the storage slot as i described above fixes the issue.

cc @IGI-111 @esdrubal @Voxelot

**Comment by IGI-111** [comment]
This is expected breakage. Which is why we added that syntax that allows you to pick a specific address.

**Review by IGI-111 — APPROVED** [review]

**Review by sdankel — APPROVED** [review]

**Comment by Voxelot** [comment]
cc @luizstacio to review this change, as I believe he had hexens audit the pre-existing proxy contract.

**Comment by kayagokalp** [comment]
Just to clarify,

I did not pinned owner proxy address like @esdrubal  example while building the contract as reading from it is not hardcoded and still works after changing the storage slot keys. 

To touch the audited contract in a minimal way that still gets things passing and working, I just pinned target field, like I showed in my Solution section. This can also be seen from the storage slots json file. Just wanted to make it clear.

**Review by calldelegation — APPROVED** [review]

File: sway-lib-std/src/ecr.sw
Comment (IGI-111) [reply sway-lib-std/src/ecr.sw:166]: We should probably update the standard library to expose the variable length, but at least this is backwards compatible for now.

**Comment by luizstacio** [comment]
@esdrubal @K1-R1 from what I understand we need yet to implement the changes at https://github.com/FuelLabs/sway-standard-implementations, correct?

**Comment by luizstacio** [comment]
@K1-R1 once this PR is merged we need to add these changes on `sway-standard-implementations` and re-open a PR back to update the binary if they differ.

I understand this needs to be in this way as this is a cycle dependency. As the correct flow would be to update first `sway-standard-implementations` and then here, but in this case we would also need this version to be released. 🔄

But from a risk management perspective, as we want to incentivize developers to use our audited proxy. 
We should add more safe-guards on the tooling side to prevent diffs between the binaries. By fetching them from GitHub using release tags with fixed versions or at least verifying the hash before deploy time.

@Voxelot thanks for tagging, we are going to account for the audit diffs.

**Comment by K1-R1** [comment]
> @K1-R1 once this PR is merged we need to add these changes on `sway-standard-implementations` and re-open a PR back to update the binary if they differ.
> 
> I understand this needs to be in this way as this is a cycle dependency. As the correct flow would be to update first `sway-standard-implementations` and then here, but in this case we would also need this version to be released. 🔄
> 
> But from a risk management perspective, as we want to incentivize developers to use our audited proxy. We should add more safe-guards on the tooling side to prevent diffs between the binaries. By fetching them from GitHub using release tags with fixed versions or at least verifying the hash before deploy time.
> 
> @Voxelot thanks for tagging, we are going to account for the audit diffs.

Yh I had already prepared the branch with the contract storage layout, but I cannot merge it until forc releases with https://github.com/FuelLabs/sway/pull/6366 as it fixes an issue that broke the configurables, and I believe that I will also need https://github.com/FuelLabs/fuels-rs/issues/1475 to be resolved before I can have the tests working for the contract. Then we can re-audit and use that contract in this repo 👍

**Comment by kayagokalp** [comment]
> fetching

Just to make sure we are on the same page, forc does not fetch anything online related to the proxy, it is embedded in the code in compile time. So unless we open a pr explicitly changing the file content and cut a new release and users migrate to that new release we cannot alter the proxy contract used

@luizstacio

**Comment by luizstacio** [comment]
> > fetching
> 
> Just to make sure we are on the same page, forc does not fetch anything online related to the proxy, it is embedded in the code in compile time. So unless we open a pr explicitly changing the file content and cut a new release and users migrate to that new release we cannot alter the proxy contract used we designed this such that the bin does not change at all without explicit changes applied
> 
> @luizstacio

Yes I'm aware we don't do it right now. The idea is to add such check and use the sway-standard-implementations as the source of true. And remove the manual process for ensuring security.

**Pull request merged**
